server/src/main/java/org/elasticsearch/search/aggregations/bucket/geogrid/GeoGridTiler.java

```diff
@@ -156,10 +157,37 @@ public long encode(double x, double y, int precision) {
             return GeoTileUtils.longEncode(x, y, precision);
         }
 
+        /**
+         * Sets the values of the long[] underlying {@link GeoShapeCellValues}.
+         *
+         * If the shape resides between <code>GeoTileUtils.LATITUDE_MASK</code> and 90 degree latitudes, then
+         * the shape is not accounted for since geo-tiles are only defined within those bounds.
+         *
+         * @param values     the bucket values
+         * @param geoValue   the input shape
+         * @param precision  the tile zoom-level
+         *
+         * @return the number of tiles set by the shape
+         */
         @Override
         public int setValues(GeoShapeCellValues values, MultiGeoValues.GeoValue geoValue, int precision) {
             MultiGeoValues.BoundingBox bounds = geoValue.boundingBox();
             assert bounds.minX() <= bounds.maxX();
+
+            if (precision == 0) {
+                values.resizeCell(1);
+                values.add(0, GeoTileUtils.longEncodeTiles(0, 0, 0));
+                return 1;
+            }
+
+            // geo tiles are not defined at the extreme latitudes due to them
+            // tiling the world as a square.
+            if ((bounds.top > GeoTileUtils.LATITUDE_MASK && bounds.bottom > GeoTileUtils.LATITUDE_MASK)
```

Comment:
I believe this can be improved, maybe there can be a way to wrap the shape such that all the latitudes above the map can be normalized down to the edge tile

Reply:
Not sure what is the right answer, but if the shape is out of the grid bounds, I think returning an empty intersection is the correct answer.

Reply:
makes sense